assets/javascripts/modules/checkout/validations.js

```diff
@@ -28,21 +38,49 @@ define(['$', 'modules/checkout/form-elements', 'modules/checkout/regex'], functi
     }
 
     var validatePersonalDetails = function () {
-        var emptyFields = mandatoryFieldsPersonalDetails.filter(function (field) {
-            var isEmpty = field.input.val() === '';
-            toggleError(field.container, isEmpty);
-            return isEmpty;
+
+        var staticValidations = new Promise(function (onSuccess, onFailure){
+            var emptyFields = mandatoryFieldsPersonalDetails.filter(function (field) {
+                var isEmpty = field.input.val() === '';
+                toggleError(field.container, isEmpty);
+                return isEmpty;
+            });
+            var noEmptyFields = emptyFields.length === 0;
+
+            var validEmail = regex.isValidEmail(form.$EMAIL.val());
+            renderEmailError(!validEmail);
+            toggleError(form.$EMAIL_CONTAINER, !validEmail);
```

Comment:
Another option might be, instead on performing side-effects inside the promise you pass an object to the `resolve` method with validation information about each field?